### ISSUE TITLE
BMSUserErrorDataSourceObjectNotFound error with azurerm_backup_protected_file_share

### ISSUE THREAD

**Issue by SuSaiGit** [terraform-providers/terraform-provider-azurerm#9452]
I tried the example codes below URL to setup backup for Azure storage account file share https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/backup_protected_file_share
But got below error
```
Error: Error creating/udpating Azure File Share backup item "AzureFileShare;example-share" (Vault "tfex-recovery-vault"): backup.ProtectedItemsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="BMSUserErrorDataSourceObjectNotFound" Message="Item not found.\r\nSelected item might have been deleted. Please check if item is registered to the vault."

  on main.tf line 51, in resource "azurerm_backup_protected_file_share" "share1":
  51: resource "azurerm_backup_protected_file_share" "share1" {
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
> terraform -v
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/azurerm v2.37.0
```

Here are my codes. Everything in a single main.tf file just for test purpose.
```
provider "azurerm" {
  features{}
}

resource "azurerm_resource_group" "rg" {
  name     = "tfex-recovery_vault"
  location = "West US"
}

resource "azurerm_recovery_services_vault" "vault" {
  name                = "tfex-recovery-vault"
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
  sku                 = "Standard"
}

resource "azurerm_storage_account" "sa" {
  name                     = "tfextestsa"
  location                 = azurerm_resource_group.rg.location
  resource_group_name      = azurerm_resource_group.rg.name
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_storage_share" "example" {
  name                 = "example-share"
  storage_account_name = azurerm_storage_account.sa.name
}

resource "azurerm_backup_container_storage_account" "protection-container" {
  resource_group_name = azurerm_resource_group.rg.name
  recovery_vault_name = azurerm_recovery_services_vault.vault.name
  storage_account_id  = azurerm_storage_account.sa.id
}

resource "azurerm_backup_policy_file_share" "example" {
  name                = "tfex-recovery-vault-policy"
  resource_group_name = azurerm_resource_group.rg.name
  recovery_vault_name = azurerm_recovery_services_vault.vault.name

  backup {
    frequency = "Daily"
    time      = "23:00"
  }

  retention_daily {
    count = 10
  }
}

resource "azurerm_backup_protected_file_share" "share1" {
  resource_group_name       = azurerm_resource_group.rg.name
  recovery_vault_name       = azurerm_recovery_services_vault.vault.name
  source_storage_account_id = azurerm_backup_container_storage_account.protection-container.storage_account_id
  source_file_share_name    = azurerm_storage_share.example.name
  backup_policy_id          = azurerm_backup_policy_file_share.example.id
}
```

### Affected Resource(s)

* `azurerm_backup_protected_file_share`
* `azurerm_backup_policy_file_share`
* `azurerm_backup_container_storage_account`
* `azurerm_storage_share`
* `azurerm_storage_account`
* `azurerm_recovery_services_vault`
* `azurerm_resource_group`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
No backend configuration.

### Debug Output

```
> terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_backup_container_storage_account.protection-container will be created
  + resource "azurerm_backup_container_storage_account" "protection-container" {
      + id                  = (known after apply)
      + recovery_vault_name = "tfex-recovery-vault"
      + resource_group_name = "tfex-recovery_vault"
      + storage_account_id  = (known after apply)
    }

  # azurerm_backup_policy_file_share.example will be created
  + resource "azurerm_backup_policy_file_share" "example" {
      + id                  = (known after apply)
      + name                = "tfex-recovery-vault-policy"
      + recovery_vault_name = "tfex-recovery-vault"
      + resource_group_name = "tfex-recovery_vault"
      + timezone            = "UTC"

      + backup {
          + frequency = "Daily"
          + time      = "23:00"
        }

      + retention_daily {
          + count = 10
        }
    }

  # azurerm_recovery_services_vault.vault will be created
  + resource "azurerm_recovery_services_vault" "vault" {
      + id                  = (known after apply)
      + location            = "westus"
      + name                = "tfex-recovery-vault"
      + resource_group_name = "tfex-recovery_vault"
      + sku                 = "Standard"
      + soft_delete_enabled = true
    }

  # azurerm_resource_group.rg will be created
  + resource "azurerm_resource_group" "rg" {
      + id       = (known after apply)
      + location = "westus"
      + name     = "tfex-recovery_vault"
    }

  # azurerm_storage_account.sa will be created
  + resource "azurerm_storage_account" "sa" {
      + access_tier                      = (known after apply)
      + account_kind                     = "StorageV2"
      + account_replication_type         = "LRS"
      + account_tier                     = "Standard"
      + allow_blob_public_access         = false
      + enable_https_traffic_only        = true
      + id                               = (known after apply)
      + is_hns_enabled                   = false
      + large_file_share_enabled         = (known after apply)
      + location                         = "westus"
      + min_tls_version                  = "TLS1_0"
      + name                             = "tfextestsa"
      + primary_access_key               = (sensitive value)
      + primary_blob_connection_string   = (sensitive value)
      + primary_blob_endpoint            = (known after apply)
      + primary_blob_host                = (known after apply)
      + primary_connection_string        = (sensitive value)
      + primary_dfs_endpoint             = (known after apply)
      + primary_dfs_host                 = (known after apply)
      + primary_file_endpoint            = (known after apply)
      + primary_file_host                = (known after apply)
      + primary_location                 = (known after apply)
      + primary_queue_endpoint           = (known after apply)
      + primary_queue_host               = (known after apply)
      + primary_table_endpoint           = (known after apply)
      + primary_table_host               = (known after apply)
      + primary_web_endpoint             = (known after apply)
      + primary_web_host                 = (known after apply)
      + resource_group_name              = "tfex-recovery_vault"
      + secondary_access_key             = (sensitive value)
      + secondary_blob_connection_string = (sensitive value)
      + secondary_blob_endpoint          = (known after apply)
      + secondary_blob_host              = (known after apply)
      + secondary_connection_string      = (sensitive value)
      + secondary_dfs_endpoint           = (known after apply)
      + secondary_dfs_host               = (known after apply)
      + secondary_file_endpoint          = (known after apply)
      + secondary_file_host              = (known after apply)
      + secondary_location               = (known after apply)
      + secondary_queue_endpoint         = (known after apply)
      + secondary_queue_host             = (known after apply)
      + secondary_table_endpoint         = (known after apply)
      + secondary_table_host             = (known after apply)
      + secondary_web_endpoint           = (known after apply)
      + secondary_web_host               = (known after apply)

      + blob_properties {
          + cors_rule {
              + allowed_headers    = (known after apply)
              + allowed_methods    = (known after apply)
              + allowed_origins    = (known after apply)
              + exposed_headers    = (known after apply)
              + max_age_in_seconds = (known after apply)
            }

          + delete_retention_policy {
              + days = (known after apply)
            }
        }

      + identity {
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = (known after apply)
        }

      + network_rules {
          + bypass                     = (known after apply)
          + default_action             = (known after apply)
          + ip_rules                   = (known after apply)
          + virtual_network_subnet_ids = (known after apply)
        }

      + queue_properties {
          + cors_rule {
              + allowed_headers    = (known after apply)
              + allowed_methods    = (known after apply)
              + allowed_origins    = (known after apply)
              + exposed_headers    = (known after apply)
              + max_age_in_seconds = (known after apply)
            }

          + hour_metrics {
              + enabled               = (known after apply)
              + include_apis          = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
            }

          + logging {
              + delete                = (known after apply)
              + read                  = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
              + write                 = (known after apply)
            }

          + minute_metrics {
              + enabled               = (known after apply)
              + include_apis          = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
            }
        }
    }

  # azurerm_storage_share.example will be created
  + resource "azurerm_storage_share" "example" {
      + id                   = (known after apply)
      + metadata             = (known after apply)
      + name                 = "example-share"
      + quota                = 5120
      + resource_manager_id  = (known after apply)
      + storage_account_name = "tfextestsa"
      + url                  = (known after apply)
    }

Plan: 6 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes 

azurerm_resource_group.rg: Creating...
azurerm_resource_group.rg: Creation complete after 1s [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault]
azurerm_recovery_services_vault.vault: Creating...
azurerm_storage_account.sa: Creating...
azurerm_recovery_services_vault.vault: Creation complete after 7s [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault]
azurerm_backup_policy_file_share.example: Creating...
azurerm_storage_account.sa: Still creating... [10s elapsed]
azurerm_backup_policy_file_share.example: Still creating... [10s elapsed]
azurerm_storage_account.sa: Still creating... [20s elapsed]
azurerm_backup_policy_file_share.example: Creation complete after 14s [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupPolicies/tfex-recovery-vault-policy]
azurerm_storage_account.sa: Creation complete after 22s [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.Storage/storageAccounts/tfextestsa]
azurerm_backup_container_storage_account.protection-container: Creating...
azurerm_storage_share.example: Creating...
azurerm_storage_share.example: Creation complete after 1s [id=https://tfextestsa.file.core.windows.net/example-share]
azurerm_backup_container_storage_account.protection-container: Still creating... [10s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [20s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [30s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [40s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [50s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m0s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m10s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m20s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m30s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m40s elapsed]
azurerm_backup_container_storage_account.protection-container: Still creating... [1m50s elapsed]
azurerm_backup_container_storage_account.protection-container: Creation complete after 1m59s [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupFabrics/Azure/protectionContainers/StorageContainer;storage;tfex-recovery_vault;tfextestsa]

Apply complete! Resources: 6 added, 0 changed, 0 destroyed.
C:\tmp\terraform-azure-backup-protected-file-share> terraform apply
azurerm_resource_group.rg: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault]
azurerm_recovery_services_vault.vault: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault]
azurerm_storage_account.sa: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.Storage/storageAccounts/tfextestsa]
azurerm_backup_policy_file_share.example: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupPolicies/tfex-recovery-vault-policy]
azurerm_backup_container_storage_account.protection-container: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupFabrics/Azure/protectionContainers/StorageContainer;storage;tfex-recovery_vault;tfextestsa]
azurerm_storage_share.example: Refreshing state... [id=https://tfextestsa.file.core.windows.net/example-share]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_backup_protected_file_share.share1 will be created
  + resource "azurerm_backup_protected_file_share" "share1" {
      + backup_policy_id          = "/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupPolicies/tfex-recovery-vault-policy"
      + id                        = (known after apply)
      + recovery_vault_name       = "tfex-recovery-vault"
      + resource_group_name       = "tfex-recovery_vault"
      + source_file_share_name    = "example-share"
      + source_storage_account_id = "/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.Storage/storageAccounts/tfextestsa"
    }

Plan: 1 to add, 0 to change, 0 to destroy.
C:\tmp\terraform-azure-backup-protected-file-share> terraform -v
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/azurerm v2.37.0
C:\tmp\terraform-azure-backup-protected-file-share> terraform apply
azurerm_resource_group.rg: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault]
azurerm_recovery_services_vault.vault: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault]
azurerm_storage_account.sa: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.Storage/storageAccounts/tfextestsa]
azurerm_backup_policy_file_share.example: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupPolicies/tfex-recovery-vault-policy]
azurerm_backup_container_storage_account.protection-container: Refreshing state... [id=/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupFabrics/Azure/protectionContainers/StorageContainer;storage;tfex-recovery_vault;tfextestsa]
azurerm_storage_share.example: Refreshing state... [id=https://tfextestsa.file.core.windows.net/example-share]

An execution plan has been generated and is shown below.  
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_backup_protected_file_share.share1 will be created
  + resource "azurerm_backup_protected_file_share" "share1" { 
      + backup_policy_id          = "/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.RecoveryServices/vaults/tfex-recovery-vault/backupPolicies/tfex-recovery-vault-policy"
      + id                        = (known after apply)
      + recovery_vault_name       = "tfex-recovery-vault"
      + resource_group_name       = "tfex-recovery_vault"
      + source_file_share_name    = "example-share"
      + source_storage_account_id = "/subscriptions/8208846c-576d-4c45-bd85-08a732f45db6/resourceGroups/tfex-recovery_vault/providers/Microsoft.Storage/storageAccounts/tfextestsa"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azurerm_backup_protected_file_share.share1: Creating...
azurerm_backup_protected_file_share.share1: Still creating... [10s elapsed]
azurerm_backup_protected_file_share.share1: Still creating... [20s elapsed]
azurerm_backup_protected_file_share.share1: Still creating... [30s elapsed]
azurerm_backup_protected_file_share.share1: Still creating... [40s elapsed]

Error: Error creating/udpating Azure File Share backup item "AzureFileShare;example-share" (Vault "tfex-recovery-vault"): backup.ProtectedItemsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="BMSUserErrorDataSourceObjectNotFound" Message="Item not found.\r\nSelected item might have been deleted. Please check if item is registered to the vault."

  on main.tf line 51, in resource "azurerm_backup_protected_file_share" "share1":
  51: resource "azurerm_backup_protected_file_share" "share1" {
```

### Expected Behaviour

Resources are created without error

### Actual Behaviour

Got below error
```
Error: Error creating/udpating Azure File Share backup item "AzureFileShare;example-share" (Vault "tfex-recovery-vault"): backup.ProtectedItemsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="BMSUserErrorDataSourceObjectNotFound" Message="Item not found.\r\nSelected item might have been deleted. Please check if item is registered to the vault."

  on main.tf line 51, in resource "azurerm_backup_protected_file_share" "share1":
  51: resource "azurerm_backup_protected_file_share" "share1" {
```

### Steps to Reproduce

1. `terraform apply`

### References
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/backup_protected_file_share

* #0000

**Comment by mcgirr** [comment]
Is there any update on this issue? I'm encountering what seems to be the same bug in the provider.

**Comment by Dhishyanth** [comment]
Even Iam also getting the same error

**Comment by awidner** [comment]
We are not even able to set this up manually in our subscription. Currently have a ticket open with Microsoft who've escalated.

**Comment by kibnelbachyr** [comment]
Any updates on this issue? I am facing the same behaviour and error during terraform apply.

**Comment by geekzter** [comment]
@kibnelbachyr I think [this PR](https://github.com/terraform-providers/terraform-provider-azurerm/pull/9015) should close the issue. It is mentioned in the [2.42 release](https://github.com/terraform-providers/terraform-provider-azurerm/milestone/108).

**Comment by mcgirr** [comment]
I just updated my project to the `2.42.0` release and tried it, but it doesn't seem to have resolved the issue.

**Comment by geekzter** [comment]
It's the 2.43 release

**Comment by geekzter** [comment]
It keeps on being moved to whatever the next release is, I stopped making assumptions.

**Comment by aristosvo** [comment]
It is merged, so next release it is!🎉🎉

Thanks @katbyte, it's amazing work you and your team is doing!

**Comment by aristosvo** [comment]
@SuSaiGit Can you confirm the issue is fixed in release `2.44.0`

**Comment by kibnelbachyr** [comment]
I just tested with the 2.44.0 release this afternoon and it works!

**Comment by katbyte** [comment]
closing as this seems to have been fixed in v2.44
